crds/workspace.devfile.io_devworkspaces.yaml

```diff
@@ -4605,6 +4612,118 @@ spec:
+                              properties:
+                                args:
+                                  description: The arguments to supply to the dockerfile
+                                    build.
+                                  items:
+                                    type: string
+                                  type: array
+                                buildContext:
+                                  description: Path of source directory to establish
+                                    build context. Defaults to ${PROJECT_ROOT} in
+                                    the container
+                                  type: string
+                                git:
+                                  description: Dockerfile's Git source
+                                  properties:
+                                    checkoutFrom:
+                                      description: Defines from what the project should
+                                        be checked out. Required if there are more
+                                        than one remote configured
+                                      properties:
+                                        remote:
+                                          description: The remote name should be used
+                                            as init. Required if there are more than
+                                            one remote configured
+                                          type: string
+                                        revision:
+                                          description: The revision to checkout from.
+                                            Should be branch name, tag or commit id.
+                                            Default branch is used if missing or specified
+                                            revision is not found.
+                                          type: string
+                                      type: object
+                                    gitLocation:
+                                      description: Location of the Dockerfile in the
+                                        Git repository when using git as Dockerfile
+                                        src.
+                                      type: string
+                                    remotes:
+                                      additionalProperties:
+                                        type: string
+                                      description: The remotes map which should be
+                                        initialized in the git project. Must have
+                                        at least one remote configured
+                                      type: object
+                                  required:
+                                  - remotes
+                                  type: object
+                                locationType:
```

Comment:
Could you elaborate why it's needed? If I understand correctly, it's going to be:
```yaml
components:
  - image:
      dockerimage:
        locationType: git
        git:
        ....
```
Do I miss something?

Reply:
This is the union discriminator. 
![Screen Shot 2021-09-08 at 11 48 29 AM](https://user-images.githubusercontent.com/31771087/132542281-a47d8fb9-003a-4754-be29-b09c01be03b6.png)


Here is an example of how it looks like with a Kubernetes component src:

![Screen Shot 2021-09-08 at 11 48 48 AM](https://user-images.githubusercontent.com/31771087/132542285-1668726b-8ec5-43c1-9548-65de12ae1b0e.png)

Reply:
Yeah, it's used for merging/overriding -- if the discriminator doesn't match we need to replace rather than end up with e.g. a component with both a `volume` and `container` field.